source/dub/generators/build.d

```diff
@@ -565,3 +565,35 @@ private Path getTargetPath(in ref BuildSettings bs, in ref GeneratorSettings set
 {
 	return Path(bs.targetPath) ~ settings.compiler.getTargetFileName(bs, settings.platform);
 }
+
+
+unittest { // issue #1235 - pass no library files to compiler command line when building a static lib
+	import dub.internal.vibecompat.data.json : parseJsonString;
+	import dub.compilers.gdc : GDCCompiler;
+
+	auto desc = parseJsonString(`{"name": "test", "targetType": "library", "sourceFiles": ["foo.d", "bar.a"]}`);
+	auto pack = new Package(desc, Path("/tmp/fooproject"));
+	auto pman = new PackageManager(Path("/tmp/foo/"), Path("/tmp/foo/"), false);
+	auto prj = new Project(pman, pack);
+
+	final static class TestCompiler : GDCCompiler {
+		override void invoke(in BuildSettings settings, in BuildPlatform platform, void delegate(int, string) output_callback) {
+			assert(!settings.dflags[].any!(f => f.endsWith("bar.a")));
```

Comment:
canFind would be safer here.

Reply:
Yes. For example if the Generator adds an extra whitespace `endsWith("bar.a")` would be true..

Reply:
Hm... can you give a full example of what case you have in mind and how it would go wrong? I guess shouldn't have asked an "or" question ;)

Reply:
Buggy implementation returns `["bar.a "]` - sorry for being so vague. I just wanted to say that using `canFind` is safer for the future.

Reply:
Mh.. okay :-)